contracts/vaults/Swap.sol

```diff
@@ -62,23 +62,24 @@ contract Swap is Base {
     }
 
     function swapTokenForHAT(uint256 _amount,
-                            IERC20 _token,
-                            uint256 _amountOutMinimum,
-                            address _routingContract,
-                            bytes calldata _routingPayload)
+        IERC20 _token,
+        uint256 _amountOutMinimum,
+        address _routingContract,
+        bytes calldata _routingPayload)
     internal
     returns (uint256 swapTokenReceived)
     {
         if (address(_token) == address(swapToken)) {
             return _amount;
         }
-
+        require(whitelistedRouters[_routingContract], "HVE44");
         require(_token.approve(_routingContract, _amount), "HVE31");
         uint256 balanceBefore = swapToken.balanceOf(address(this));
+        require(balanceBefore >= _amountOutMinimum, "HVE45");
```

Comment:
?????

Reply:
checking that we have enough rewards to do the swap, what's wrong with that?